File: backend/rust/services/draft/src/schemas.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct DraftCommand(pub Team, pub Action, pub Option<usize>);

#[derive(Serialize, Deserialize, Clone, PartialEq, Eq)]
pub enum Team {
    Red,
    Blue,
}
#[derive(Serialize, Deserialize, Clone, PartialEq, Eq)]
pub enum Action {
    Pick,
    Ban,
}

File: backend/python/shared/src/shared/contracts/draft/schemas.py
from enum import Enum

from pydantic import BaseModel

from uuid import UUID


class Team(BaseModel):
    type: TeamType
    id: UUID


class TeamType(Enum):
    RED = "red"
    BLUE = "blue"


class Action(BaseModel):
    type: ActionType
    champion_id: int


class ActionType(Enum):
    PICK = "pick"
    BAN = "ban"


class Command(BaseModel):
    team: Team
    action: Action


class PrepareDraft(BaseModel):
    game_id: UUID
    forbidden_champions: list[int]
    teams: list[Team]
    seconds_per_action: int
    team_size: int
